crypto/hmac/hmac.c

```diff
@@ -37,7 +37,8 @@ int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
     if (key != NULL) {
         reset = 1;
         j = EVP_MD_block_size(md);
-        OPENSSL_assert(j <= (int)sizeof(ctx->key));
+        if (!ossl_assert(j <= (int)sizeof(ctx->key)))
+            goto err;
```

Comment:
OK, as practical example if one attempts to pass EVP_sha3_256() then it would be appropriate to return error. In other words dismiss the comment, sorry about noise. Yet on side note one can wonder if it would be appropriate to ERR_put_error something... Well, crypto/hmac doesn't do ERR...